### ISSUE TITLE
More units of measure

### ISSUE THREAD

**Issue by bender2k14** [louthy/language-ext#421]
Quoting @thehoglet from #419 

> Any idea if units of measure is going to be further developed? Is the system just an experiment? This morning, I implemented Force, Pressure, and Temperature for a project I'm working on. I included only the conversions I needed, though. It worked nicely and Units of Measure improved the API I'm working on. But, the experience informed me that implementing this for the entire universe of SI+conversions is a big job!

I hope it will be further developed.  I also would benefit from some additional units of measure.  The units of measure that I want are for measuring the size of a file or the capacity of a directory as well as for specifying the rate of a data transfer.

I have put some though into what names to use for these two units of measure.  I currently like the name `Information` and `InformationRate`.  The first sentence from the [Wikipedia article for Information theory](https://en.wikipedia.org/wiki/Information_theory) says
> Information theory studies the quantification, storage, and communication of [information](https://en.wikipedia.org/wiki/Information).

The size of a file is an example of quantification, the capacity of a directory is an example of storage, and the rate of a data transfer is an example of communication.

**Comment by thehoglet** [comment]
If desired, I can incorporate my implementations of Force, Pressure and Temperature plus one for Mass (plus unit tests) into language-ext and issue a pull request. Let me know.

(In implementing a UoM for force, I omitted any mention of mass. I'm guessing the best return for the multiplication operator for the language-ext Accel type, if the LHS/RHS is mass, is a Force instance. Also, there seems to be a proposal to alter the SI definition of kelvin that will introduce a dependency on mass. One way or another the lib appears to need the mass SI base unit implemented.)

**Comment by louthy** [comment]
> If desired, I can incorporate my implementations of Force, Pressure and Temperature plus one for Mass (plus unit tests) into language-ext and issue a pull request. Let me know.

Yes please

> (In implementing a UoM for force, I omitted any mention of mass. I'm guessing the best return for the multiplication operator for the language-ext Accel type, if the LHS/RHS is mass, is a Force instance

I'm not sure I'm following what you're saying here.  The return value for the operators should be based on the operands.  So if you multiply an `Accel` by a scalar value (a `double` for example) then you should just get an `Accel` back, `Accel x Accel` should be `AccelSq`, etc.  If you divide an `Accel` by an `Accel` then you'd get a non-units value out of it (`double`).  

In all instances it's better to not provide the operator than provide one with a half-arsed solution.

**Comment by louthy** [comment]
@bender2k14 

> I have put some though into what names to use for these two units of measure. I currently like the name Information and InformationRate. The first sentence from the Wikipedia article for Information theory says
> Information theory studies the quantification, storage, and communication of information.
> The size of a file is an example of quantification, the capacity of a directory is an example of storage, and the rate of a data transfer is an example of communication.

I like the purity of what you're suggesting (name-wise), and the idea in general, but do you think names like `Information` are just a bit too vague sounding for most users?  Is there scope here for coming up with something more descriptive (without too much penalty in terms of the length of the unit name)?

**Comment by IDisposable** [comment]
More used to seeing number and dimension... http://www.mne.psu.edu/cimbala/Learning/General/units.htm

**Comment by bender2k14** [comment]
> do you think names like Information are just a bit too vague sounding for most users? Is there scope here for coming up with something more descriptive (without too much penalty in terms of the length of the unit name)?

I am definitely open to considering alternatives.  I do think it sounds rather unusual at first, but anything else that I have considered sounds even more strange to me.

I looked around today for more examples of how people measure these quantities.  I only found more evidence that `Information` is the correct name.  On Wikipedia for example, in addition to articles called [Unit of length](https://en.wikipedia.org/wiki/Unit_of_length) and [Unit of time](https://en.wikipedia.org/wiki/Unit_of_time) is an article called [Units of information](https://en.wikipedia.org/wiki/Units_of_information).  It lists the units that I would expect to see such as "bit", "byte", the powers of 10 like "kilobit" or "kilobyte", and the powers of 2 like "kibibit" or "kilibyte.

If we accept `Information`, there is still some uncertainly regarding `InformationRate`.  In the context of measurements, "rate" means (in the most general sense) a ratio of two measurements.  In my experience, such rates almost always have a measurement of time in the denominator.  Examples include growth rate (where the numerator might be a currency, the size of population, or the size of a single object measured in length, area, or volume).  One counterexample to this pattern is exchange rate, which is a ratio of two different currencies.

However, we should also consider what types of measurements one by want to divide into information.  The only such example from the [Wikipedia article for Information theory](https://en.wikipedia.org/wiki/Information_theory) is information over time.  I know of one other example though.  Some parts of physics consider the density of information, which would be information over volume.  If someone had a need to model this programmatically though, I would think that `InformationDensity` would be fine name.

So, in conclusion, my preference is still for `Information` and `InformationRate`.

**Comment by louthy** [comment]
> So, in conclusion, my preference is still for Information and InformationRate.

Yes, with a bit more thought I'm ok with this.  Especially as it will be in the UnitsOfMeasure namespace and is therefore unlikely to cause clashes with users' types.
